org-eclipse-che-jdt-ls-extension/jdt.ls.extension.api/src/main/java/org/eclipse/che/jdt/ls/extension/api/dto/JavaProjectStructure.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.jdt.ls.extension.api.dto;
+
+import java.util.List;
+
+/**
+ * Represents Java project in terms of JavaModel.
+ *
+ * @author Valeriy Svydenko
+ */
+public class JavaProjectStructure {
```

Comment:
I think structural elements could be more generic, of the kind:
```
Element {
  kind: Kind;
  name: String;
  uri: String;
  children?: Element[];
}

Reply:
But I think it's probably better if we merge the PR like it is and refactor later. The better is the enemy of the good :-)

Reply:
created an issue https://github.com/eclipse/che/issues/10045